src/state/slices/foxEthSlice/utils.ts

```diff
@@ -0,0 +1,15 @@
+import { Fetcher, Token } from '@uniswap/sdk'
+import { providers } from 'ethers'
+import memoize from 'lodash/memoize'
+import { getEthersProvider } from 'plugins/foxPage/utils'
+
+export const ethersProvider = getEthersProvider()
+
+export const fetchPairData = memoize(
+  async (
+    tokenA: Token,
+    tokenB: Token,
+    fetchPairData: typeof Fetcher['fetchPairData'],
+    provider: providers.Web3Provider,
```

Comment:
gut feel is this won't properly memoize and we'll get diff references on diff call, please check - might need a custom equality function here or rethink the API